internal/faustjs.org/docs/gutenberg/how-to-query-blocks.mdx

```diff
@@ -0,0 +1,254 @@
+---
+slug: how-to-query-blocks
+title: Various strategies to query block data attributes using wp-graphql-content-blocks plugin
+---
+
+# How to query blocks
+
+The WPGraphQL Content Blocks plugin extends the WPGraphQL schema with new graphql types and fields.
+This how to guide gives a brief overview of the different query strategies you can use.
+
+## EditorBlock
+
+* An interface type which represents a single block that is available in the block editor.
+* A base type for all other object block types.
+
+Here is an example query:
+
+```graphql
+fragment Paragraph on CoreParagraph {
+  	attributes {
+    	content
+  	}
+}
+
+query GetAllPostsWhichSupportBlockEditor {
+	posts {
+		edges {
+			node {
+				# editorBlocks field represents array of EditorBlock
+				editorBlocks {
+					# fields from the interface
+					__typename
+					clientId
+  					parentClientId
+					name
+					...Paragraph
+				}
+			}
+		}
+	}
+}
+```
+
+## NodeWithEditorBlocks
+
+* An interface type which represents a list of `EditorBlock` types.
+
+Here is an example query:
+
+```graphql
+fragment Paragraph on CoreParagraph {
+  	attributes {
+    	content
+  	}
+}
+
+query GetAllPostsWhichSupportBlockEditor {
+	nodeByUri(uri: "/posts/hello-world") {
+    ... on NodeWithEditorBlocks {
+      editorBlocks {
+        __typename
+        clientId
+        parentClientId
+        name
+        ...Paragraph
+      }
+    }
+  }
+}
+```
+
+and here is the result:
+
+```json
+{
+	"data": {
+		"nodeByUri": {
+			"editorBlocks": [
+				{
+					"__typename": "CoreParagraph",
+					"clientId": "642d960f45c05",
+					"parentClientId": null,
+					"name": "core/paragraph",
+					"attributes": {
+						"content": "Welcome to WordPress. This is your first post. Edit or delete it, then start writing!"
+					}
+				}
+			]
+		}
+  	},
+}
+```
+
+# How to query posts associated with a specific Post type
+
+The plugin also offers support for the [allowed_block_types_all](https://developer.wordpress.org/reference/hooks/allowed_block_types_all/)
+filter which controls which blocks are allowed in the block editor.
+
+If you set a filter that returns a different list of blocks per post type the plugin will register a more specific set of interfaces for each post type.
+
```

Comment:
I got a little confused here, so I've made a small visual for this section if you'd like to add it- it just runs the below code and displays the differences pre- and post- filter:

https://drive.google.com/file/d/1grv_GFIY2bBrXiMZI8Uoa0f0s8pfb4E_/view?usp=sharing

Reply:
Hey @TeresaGobble Thats cool. I will add this video recording there.

Reply:
Oops its 39mb in size it would it be hard to stream that? I will try